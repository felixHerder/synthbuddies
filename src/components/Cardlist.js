import React from 'react';
import Card from './Card';


const Cardlist = ({ robots }) => {
    //if(true) throw new Error("Hell no!");
    console.log('Cardlist rendered');
    return (
        <div>
            {
                robots.map(user => {
                    return <Card key={user.id} id={user.id} name={user.name} email={user.email} />
                })
            }
        </div>
    );
}

export default Cardlist;