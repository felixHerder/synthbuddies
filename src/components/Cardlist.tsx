import React from 'react';
import Card from './Card';
import { IRobot } from './MainPage';


const Cardlist = ({ robots }: { robots: Array<IRobot> }) => {
    //if(true) throw new Error("Hell no!");
    return (
        <div>
            {
                robots.map(user => {
                    return <Card key={user.id} id={user.id} name={user.name} email={user.email} />
                })
            }
        </div>
    );
}

export default Cardlist;