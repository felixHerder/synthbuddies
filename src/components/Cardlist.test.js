import { shallow, mount, render } from 'enzyme';
import React from 'react';
import Cardlist from './Cardlist';

it('expects to render Cardlist comp', () => {
  const mockRobots = [
    {
      id: 1,
      name: "John Dode",
      username: "johnjohn",
      email: 'john@email.com'
    }
  ];
  expect((shallow(<Cardlist robots={mockRobots} />))).toMatchSnapshot();
});