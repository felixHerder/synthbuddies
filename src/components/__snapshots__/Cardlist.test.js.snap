// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render Cardlist comp 1`] = `
<div>
  <Card
    email="john@email.com"
    id={1}
    key="1"
    name="John Dode"
  />
</div>
`;
